klippa/src/sbix.rs

```diff
@@ -0,0 +1,193 @@
+//! impl subset() for sbix
+use crate::serialize::{OffsetWhence, SerializeErrorFlags, Serializer};
+use crate::{Plan, Subset, SubsetError, SubsetTable};
+use skrifa::GlyphId;
+use write_fonts::types::FixedSize;
+use write_fonts::{
+    read::{
+        tables::sbix::{Sbix, Strike},
+        types::Offset32,
+        ArrayOfOffsets, FontRef, TopLevelTable,
+    },
+    FontBuilder,
+};
+
+// reference: subset() for sbix in harfbuzz
+// <https://github.com/harfbuzz/harfbuzz/blob/4df11621cecf6cf855e9e13f6f5c9432748f9b3a/src/OT/Color/sbix/sbix.hh#L420>
+impl Subset for Sbix<'_> {
+    fn subset(
+        &self,
+        plan: &Plan,
+        _font: &FontRef,
+        s: &mut Serializer,
+        _builder: &mut FontBuilder,
+    ) -> Result<(), SubsetError> {
+        serialize_header(self, s).map_err(|_| SubsetError::SubsetTableError(Sbix::TAG))?;
+        self.strikes()
+            .subset(plan, s, &())
+            .map_err(|_| SubsetError::SubsetTableError(Sbix::TAG))
+    }
+}
+
+fn serialize_header(sbix: &Sbix, s: &mut Serializer) -> Result<(), SerializeErrorFlags> {
+    s.embed(sbix.version())?;
+    s.embed(sbix.flags()).map(|_| ())
+}
+
+impl<'a> SubsetTable<'a> for ArrayOfOffsets<'a, Strike<'a>, Offset32> {
+    type ArgsForSubset = ();
+    fn subset(
+        &self,
+        plan: &Plan,
+        s: &mut Serializer,
+        _args: &Self::ArgsForSubset,
+    ) -> Result<(), SerializeErrorFlags> {
+        let num_strikes_pos = s.embed(0_u32)?;
+        let org_num = self.len();
```

Comment:
huge nit but 'org' always reads as shorthand for 'organization' to me, I'd maybe call this `orig_len`?